sktime/forecasting/compose/_category_compositor.py

```diff
@@ -0,0 +1,479 @@
+
+    Raises
+    ------
+    AssertionError: If a valid transformer (an instance of BaseTransformer)
+    is not passed or if valid forecasters (instances of BaseForecaster) are not given.
+
+    Examples
+    --------
+    >>> from sktime.forecasting.compose._category_compositor import CategoryCompositor
+    >>> from sktime.forecasting.croston import Croston
+    >>> from sktime.forecasting.trend import PolynomialTrendForecaster
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.transformations.series.adi_cv import ADICVTransformer
+    >>> from sktime.transformations.series.tests.test_adi_cv import (
+    ...     _generate_erratic_series)
+    >>> category_compositor = CategoryCompositor(
+    ...     forecasters =
+    ...         {"smooth": NaiveForecaster(),
+    ...         "erratic": Croston(),
+    ...         "intermittent": PolynomialTrendForecaster()},
+    ...     transformer=ADICVTransformer(features=["class"]))
+
+    >>> generated_data = _generate_erratic_series()
+    >>> category_compositor = category_compositor.fit(generated_data, fh=50)
+
+    >>> # Verifies that the correct forecaster is chosen
+    >>> # print(category_compositor.chosen_forecaster_)
+
+    >>> # Print out the predicted value!
+    >>> # print(category_compositor.predict(fh=50, X=None))
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "scitype:y": "both",
+        "ignores-exogeneous-X": False,
+        "requires-fh-in-fit": False,
+        "enforce_index_type": None,
+        "authors": ["shlok191"],
+        "maintainers": ["shlok191"],
+        "python_version": None,
+    }
+
+    def __init__(
+        self,
+        forecasters,
+        transformer=adi_cv_transformer,
+        fallback_forecaster=None,
+    ):
+        # saving arguments to object storage
+        self.transformer = transformer
+        self.forecasters = forecasters
+        self.fallback_forecaster = fallback_forecaster
+
+        self.transformer_ = copy.deepcopy(transformer)
+
+        super().__init__()
+
+        # validating passed arguments
+        assert isinstance(transformer, BaseTransformer)
+
+        for forecaster in forecasters.values():
+            assert isinstance(forecaster, BaseForecaster)
+
+        # All checks OK!
+
+        # Assigning all capabilities on the basis of the capabilities
+        # of the passed forecasters
+        true_if_all_tags = {
+            "ignores-exogeneous-X": False,
+            "X-y-must-have-same-index": False,
+            "enforce_index_type": False,
+            "handles-missing-data": False,
+            "capability:insample": False,
+            "capability:pred_int": False,
+            "capability:pred_int:insample": False,
+        }
+
+        # Extrapolating values for flags that should be True if they are
+        # True for all of the given forecasters!
+        for tag in true_if_all_tags.keys():
```

Comment:
this block of code is very long. Why not use `self._anytagis_then_set`?

Reply:
I actually tried out `self._anytagis_then_set`, but I got the error that the function was not defined. Is it a defined function that the class should have inherited from `BaseForecaster`?